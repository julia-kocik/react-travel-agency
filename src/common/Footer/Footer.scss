@import '../../settings.scss';

.footer_container {
    height: 100px;
    display: flex;
    justify-content: center;
    align-items: center;
    color: $color-white;
    background-color: $color-blue;
    background: $color-gradient;
    background: $linear-gradient;
    span {
        font-family: $font-main;
    }
}
