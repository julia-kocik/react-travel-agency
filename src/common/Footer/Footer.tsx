import React from 'react'
import './Footer.scss'

const Footer = () => {
  return (
    <div data-testid="footer-container" className='footer_container'><span>Julia Kocik 2023</span></div>
  )
}

export default Footer