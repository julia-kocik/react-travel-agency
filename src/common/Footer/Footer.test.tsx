import React from 'react';
import { render } from '@testing-library/react';
import Footer from './Footer';
import './Footer.scss'

describe('Footer component', () => {
  it('should render correctly', () => {
    const { container } = render(<Footer />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should contain a div with class "footer_container"', () => {
    const { getByTestId } = render(<Footer />);
    expect(getByTestId('footer-container').classList.contains('footer_container')).toBe(true);
  });

  it('should contain a span with text "Julia Kocik 2023"', () => {
    const { getByText } = render(<Footer />);
    expect(getByText('Julia Kocik 2023')).toBeInTheDocument();
  });

});
