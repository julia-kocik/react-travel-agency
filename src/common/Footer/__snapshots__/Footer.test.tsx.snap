// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render correctly 1`] = `
<div
  class="footer_container"
  data-testid="footer-container"
>
  <span>
    Julia Kocik 2023
  </span>
</div>
`;
